Automattic/wp-calypso
# WPCOM: Migrate `wpcom.undocumented()` account recovery to `wpcom.req`

#### Changes proposed in this Pull Request

This PR migrates the `wpcom.undocumented()` account recovery settings to `wpcom.req.post()` and `wpcom.req.get()`. 

Part of the ongoing effort to get rid of `wpcom.undocumented()`.

#### Testing instructions

* Go to `/me/security/account-recovery`
* Verify retrieving and saving the email and phone number there still works well.
* Verify all tests still pass: `yarn run test-client client/state/account-recovery/settings/test/actions.js`
